Sources/StreamChat/Query/UserListQuery.swift

```diff
@@ -14,46 +14,112 @@ public class UserListFilterScope: FilterScope, AnyUserListFilterScope {}
 /// Non extra-data-specific filer keys for channel list.
 public extension FilterKey where Scope: AnyUserListFilterScope {
     /// A filter key for matching the `id` value.
-    static var id: FilterKey<Scope, UserId> { "id" }
+    static var id: FilterKey<Scope, UserId> {
+        .init(
+            payloadKey: UserPayloadsCodingKeys.id,
+            dtoKey: #keyPath(UserDTO.id)
+        )
+    }
```

Comment:
Does it make sense in this PR to create the auto keys for all Controllers? Maybe it is a good idea to limit the scope for now to the Channel List, which is the most problematic one. Then we can create tickets to handle the other controllers (AFAIK we already have this for some, ex: members)